yt_dlp/extractor/facebook.py

```diff
@@ -1017,19 +1034,31 @@ def _real_extract(self, url):
         video_id = self._match_id(url)
         webpage = self._download_webpage(url, video_id)
 
-        post_data = [self._parse_json(j, video_id, fatal=False)
-                     for j in re.findall(r's\.handle\(({.*})\);requireLazy\(', webpage)]
-        data = traverse_obj(post_data, (
-            ..., 'require', ..., ..., ..., 'props', 'deeplinkAdCard', 'snapshot', {dict}), get_all=False)
+        if post_data := [self._parse_json(j, video_id, fatal=False)
+                         for j in re.findall(r'data-sjs>({.*?ScheduledServerJS.*?})</script>', webpage)]:
+            data = traverse_obj(post_data, (
+                ..., 'require', ..., ..., ..., '__bbox', 'require', ..., ..., ...,
+                'entryPointRoot', 'otherProps', 'deeplinkAdCard', 'snapshot', {dict}), get_all=False)
+        elif post_data := [self._parse_json(j, video_id, fatal=False)
+                           for j in re.findall(r's\.handle\(({.*})\);requireLazy\(', webpage)]:
+            data = traverse_obj(post_data, (
+                ..., 'require', ..., ..., ..., 'props', 'deeplinkAdCard', 'snapshot', {dict}), get_all=False)
```

Comment:
I can't find any examples that still work with the old regex pattern / traversal path. The old fallback is adding unnecessary complexity, IMO, which can lead to bugs (e.g. `data` can be unbound if `else`).

I would like to simplify it like this:
```suggestion
        post_data = traverse_obj(
            re.findall(r'data-sjs>({.*?ScheduledServerJS.*?})</script>', webpage), (..., {json.loads}))
        data = get_first(post_data, (
            'require', ..., ..., ..., '__bbox', 'require', ..., ..., ...,
            'entryPointRoot', 'otherProps', 'deeplinkAdCard', 'snapshot', {dict}))
```